SingularityBase/src/main/java/com/hubspot/mesos/MesosUtils.java

```diff
@@ -39,17 +42,44 @@ private static double getScalar(Resource r) {
     return r.getScalar().getValue();
   }
 
-  private static double getScalar(List<Resource> resources, String name) {
+  private static double getScalar(List<Resource> resources, String name, Optional<String> requiredRole) {
     for (Resource r : resources) {
 
-      if (r.hasName() && r.getName().equals(name) && r.hasScalar()) {
+      if (r.hasName() && r.getName().equals(name) && r.hasScalar() && isRequiredRole(r, requiredRole)) {
         return getScalar(r);
       }
     }
 
     return 0;
   }
 
+  private static Boolean hasRole(Resource r) {
+    return r.hasRole() && !r.getRole().equals("*");
+  }
+
+  private static Boolean isRequiredRole(Resource r, Optional<String> requiredRole) {
```

Comment:
this method name is a little confusing to me -- it determines whether or not the resource is of the desired role -- would you be open to renaming it to `hasRequiredRole()` or `isResourceMatchesRequiredRole()`?